Src/Support/Google.Apis.Auth.Tests/OAuth2/ExternalAccountCredentialTestsBase.cs

```diff
@@ -43,14 +44,125 @@ public abstract class ExternalAccountCredentialTestsBase
+            Assert.Throws<UriFormatException>(() => new DummyExternalCredential(new DummyExternalCredential.Initializer(tokenUrl, Audience, SubjectTokenType)));
+
+        [Theory]
+        [InlineData("https://iamcredentials.googleapis.com")]
+        [InlineData("https://us-east-1.iamcredentials.googleapis.com")]
+        [InlineData("https://US-EAST-1.iamcredentials.googleapis.com")]
+        [InlineData("https://iamcredentials.us-east-1.googleapis.com")]
+        [InlineData("https://iamcredentials.US-WEST-1.googleapis.com")]
+        [InlineData("https://us-east-1-iamcredentials.googleapis.com")]
+        [InlineData("https://US-WEST-1-iamcredentials.googleapis.com")]
+        [InlineData("https://us-west-1-iamcredentials.googleapis.com/path?query")]
+        [InlineData("https://iamcredentials-xyz123.p.googleapis.com/path?query")]
+        [InlineData("https://iamcredentials-xyz123.p.googleapis.com")]
+        [InlineData("https://iamcredentials-xyz-123.p.googleapis.com")]
+        public void ValidateImpersonationUrl_ValidUrl(string impersonationUrl) =>
+            new DummyExternalCredential(new DummyExternalCredential.Initializer(TokenUrl, Audience, SubjectTokenType)
+            {
+                ServiceAccountImpersonationUrl = impersonationUrl
+            });
+
+        [Theory]
+        [InlineData("https://sts.googleapis.com")]
+        [InlineData("http://iamcredentials.googleapis.com")]
+        [InlineData("https://iamcre.dentials.googleapis.com")]
+        [InlineData("https://US-WE/ST-1-iamcredentials.googleapis.com")]
+        [InlineData("https://iamcredentials-us-east-1.googleapis.com")]
+        [InlineData("https://iamcredentials-US-WEST-1.googleapis.com")]
+        [InlineData("testhttps://us-east-1.iamcredentials.googleapis.com")]
+        [InlineData("https://us-east-1.iamcredentials.googleapis.comevil.com")]
+        [InlineData("https://us-east-1.us-east-1.iamcredentials.googleapis.com")]
+        [InlineData("https://us-ea.s.t.iamcredentials.googleapis.com")]
+        [InlineData("https://iamcredentials.googleapis.comevil.com")]
+        [InlineData("hhttps://us-east-1.iamcredentials.googleapis.com")]
+        [InlineData("https://-iamcredentials.googleapis.com")]
+        [InlineData("https://us-east-1.iamcredentials.googleapis.com.evil.com")]
+        [InlineData("https://iamcredentials.pgoogleapis.com")]
+        [InlineData("https://p.googleapis.com")]
+        [InlineData("https://iamcredentials.p.com")]
+        [InlineData("http://iamcredentials.p.googleapis.com")]
+        [InlineData("https://xyz-iamcredentials.p.googleapis.com")]
+        [InlineData("https://iamcredentials-xyz.123.p.googleapis.com")]
+        [InlineData("https://iamcredentials-xyz.p1.googleapis.com")]
+        [InlineData("https://iamcredentials-xyz.p.foo.com")]
+        [InlineData("https://iamcredentials-xyz.p.foo.googleapis.com")]
+        public void ValidateImpersonationUrl_InvaliddUrl(string impersonationUrl) =>
```

Comment:
Nit - here and below - Invalidd => Invalid

Reply:
Done